Lib/test/support/strace_helper.py

```diff
@@ -71,6 +71,27 @@ def sections(self):
         return sections
 
 
+def filter_memory(syscalls):
+    """Filter out memory allocation calls from File I/O calls.
+
+    Some calls (mmap, munmap, etc) can be used on files or to just get a block
+    of memory. Use this function to filter out the memory related calls from
+    other calls."""
+
+    def _filter(call):
+        # mmap can operate on a fd or `NULL` which gives a block of memory.
+        # Ignore the `NULL` ones.
+        if call.syscall == 'mmap' and call.args[0] == 'NULL':
```

Comment:
```suggestion
        if call.syscall == 'mmap' and call.args[4] == '-1':
```

I think the key here is that they pass `-1` as fd.

Reply:
Unfortunately for `glibc` man pages at least `-1` isn't required... Going to see if I can look for `MAP_ANONYMOUS` easily. (Fortunately this code is linux only, MacOS did MAP_ANON....)


---
``` 
       MAP_ANONYMOUS
              The mapping is not backed by any file; its contents are initialized to zero.  The fd argument is ignored; however, some implementations require fd to be -1 if MAP_ANONYMOUS (or MAP_ANON) is specified, and portable applications should ensure this.  The  off‐
              set argument should be zero.  Support for MAP_ANONYMOUS in conjunction with MAP_SHARED was added in Linux 2.4.
```


https://man7.org/linux/man-pages/man2/mmap.2.html

Reply:
I don't think that's really a problem here but I suppose this is even a better approach, so :+1:.